### ISSUE TITLE
Conversions regeneration problem

### ISSUE THREAD

**Issue by Rhincodon** [spatie/laravel-medialibrary#104]
Hi, I am using version 3.10.1 and have a problem with image regenerations. When I regenerating image conversions with `medialibrary:regenerate`, I got those error and stack trace(its not full):

```
[2015-11-12 18:54:57] local.ERROR: exception 'ErrorException' with message 'finfo::buffer() expects parameter 1 to be string, resource given' in /vendor/league/flysystem/src/Util/MimeType.php:26
Stack trace:
#0 [internal function]: Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'finfo::buffer()...', '/home/vagrant/C...', 26, Array)
#1 /vendor/league/flysystem/src/Util/MimeType.php(26): finfo->buffer(Resource id #532)
#2 /vendor/league/flysystem/src/Util.php(162): League\Flysystem\Util\MimeType::detectByContent(Resource id #532)
#3 /vendor/league/flysystem/src/Adapter/Local.php(186): League\Flysystem\Util::guessMimeType('8/4/6/39/c/admi...', Resource id #532)
#4 /vendor/league/flysystem/src/Filesystem.php(96): League\Flysystem\Adapter\Local->update('8/4/6/39/c/admi...', Resource id #532, Object(League\Flysystem\Config))
#5 /vendor/spatie/laravel-medialibrary/src/Filesystem.php(73): League\Flysystem\Filesystem->put('8/4/6/39/c/admi...', Resource id #532, Array)
#6 /vendor/spatie/laravel-medialibrary/src/FileManipulator.php(82): Spatie\MediaLibrary\Filesystem->copyToMediaLibrary('/home/vagrant/C...', Object(Spatie\MediaLibrary\Media), true)
#7 /vendor/spatie/laravel-medialibrary/src/Jobs/PerformConversions.php(38): Spatie\MediaLibrary\FileManipulator->performConversions(Object(Spatie\MediaLibrary\Conversion\ConversionCollection), Object(Spatie\MediaLibrary\Media))
```

I discovered problem in this lines https://github.com/spatie/laravel-medialibrary/blob/2527698d8bcaac025b6f8dfa64becf99030f88f5/src/Filesystem.php#L72-L73
When we trying `put` directly on driver, laravel behavior for second parameter as `$resource` http://laravel.com/docs/5.1/filesystem#storing-files doesn't work, cause we skip laravels Filesystem adapter.

Deleting `getDriver()` and third parameter on `put` solve problem.

**Comment by pascalbaljet** [comment]
This problem occurs when using 3.7.2 or later, 3.7.1 is fine.

**Comment by freekmurze** [comment]
I'll take a look at this and fix it soon.

On Sunday, 15 November 2015, pascalbaljet <notifications@github.com> wrote:

> This problem occurs when using 3.7.2 or later, 3.7.1 is fine.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/spatie/laravel-medialibrary/issues/104#issuecomment-156814848>
> .
>


-- 
Freek Van der Herten
https://spatie.be
+32 495 84 27 91

**Comment by freekmurze** [comment]
This is fixed in v3.10.2 of the package.

**Comment by pascalbaljet** [comment]
Thanks!
